### PR TITLE
feat(docs): move architecture docs to Starlight docs #530

### LINKED ISSUE #530: Transfer dev/architecture docs

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #530 

and makes #157 and #158 obsolete

### CODE REVIEW

File: lapis2-docs/astro.config.mjs [hunk 1]
```diff
@@ -83,6 +83,27 @@ export default defineConfig({
                         },
                     ]),
                 },
+                {
+                    label: 'Architecture and Dev Docs',
```
Comment (JonasKellerer) [inline lapis2-docs/astro.config.mjs:87]: When I run the container locally, I cant see these links.

Comment (JonasKellerer) [reply lapis2-docs/astro.config.mjs:87]: And also the routes dont work for me

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- Developed under an open-source licence. We chose the tooling such that a broad spectrum of developers can in principle
+work on the software.
+- The software is designed to be highly configurable so that it can be used for various organisms.
+Configuration files have to be passed to LAPIS and SILO at runtime that determine the nature of the organism such as:
+  - a reference genome
+  - which metadata is available on the genomic data
+- Input files for SILO are provided in a specific format:
+  - metadata and quality control
+  - sequence data (non-aligned nucleotide sequences, aligned nucleotide sequences, aligned AA sequences)
```
Comment (JonasKellerer) [inline lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx:16]: also the pango lineage alias?

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- LAPIS is backed by a database that understands SILO queries.
+- The system is designed to have the best possible performance.
+This mostly targets SILO, but also in LAPIS,
+we have to keep in mind that we are dealing with potentially large data that we have to serve to the client.
+- We provide executable that can be run on Linux and macOS.
```
Comment (JonasKellerer) [inline lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx:21]: Do we provide them?

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/glossary.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- AA mutation from "G" to "S" at position 1234
+
+The origin AA symbol can be omitted, since it is clear from the reference genome.
+**Example**:
+ORF_1a:1234S
+
+#### Nucleotide
+
+**Example**:
+C1234T:
```
Comment (JonasKellerer) [inline lapis2-docs/src/content/docs/architecture-and-dev-docs/glossary.mdx:40]: Should we add the segments here?

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/system-scope-and-context.mdx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      - returns how many sequences fulfill the provided filter criteria
+    - amino acid mutations, nucleotide mutations
+      - returns a list of mutations (with their proportion and count) which fulfill the provided filter criteria
+      - the proportion is relative to all sequences which fulfill the provided criteria
+    - details
+      - returns a list of metadata of the sequences which fulfill the provided criteria
+    - amino acid sequences, aligned and unaligned nucleotide sequences
+      - returns the corresponding sequences which fulfill the provided criteria
+  - the provided filter criteria can be a compilation of metadata fields, mutations and insertions.
+  - LAPIS offers an OpenAPI specification and a Swagger UI that documents the API.
```
Comment (JonasKellerer) [inline lapis2-docs/src/content/docs/architecture-and-dev-docs/system-scope-and-context.mdx:27]: Do we want a link to the swagger here?

File: lapis2-docs/astro.config.mjs (hunk 1 above)
Comment (fengelniederhammer) [reply lapis2-docs/astro.config.mjs:87]: Hm, did you start it with the command from the README?
`CONFIG_FILE=../siloLapisTests/testData/testDatabaseConfig.yaml REFERENCE_GENOMES_FILE=../siloLapisTests/testData/reference_genomes.json LAPIS_URL=http://localhost:8080 npm run dev`

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx (hunk 3 above)
Comment (fengelniederhammer) [reply lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx:21]: Actually not yet. Should we simply remove this sentence?

File: lapis2-docs/src/content/docs/architecture-and-dev-docs/glossary.mdx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+| mutation | a divergence from the reference genome (see below).                                                                                                                                                                                                              |
+| Organism | The organism that the genomic data was extracted from. Each LAPIS instance serves data for a single organism.                                                                                                                                                    |
+| QC       | quality control; in our case, it usually refers to the quality checks and metrics of the sequences, targeting how well the nucleotide sequence was determined from the probe.                                                                                    |
+| segment  | TODO                                                                                                                                                                                                                                                             |
```
Comment (fengelniederhammer) [inline lapis2-docs/src/content/docs/architecture-and-dev-docs/glossary.mdx:13]: @chaoran-chen Do you have a suggestion how to best describe the segment?

**Review by JonasKellerer — APPROVED** [review]
LGTM

**Pull request merged**